src/app/static/src/tests/integration/downloadResults.itest.ts

```diff
@@ -143,10 +135,11 @@ describe(`download results actions integration`, () => {
 
         const state = {spectrum: {downloadId: 123}}
 
-        actions.poll({commit, dispatch, state, rootState: root} as any, DOWNLOAD_TYPE.SPECTRUM);
-        vi.advanceTimersByTime(3000);
-        await flushPromises();
-        expect(commit.mock.calls.length).toBe(2);
+        actions.poll({commit, dispatch, state, rootState: root} as any, DOWNLOAD_TYPE.SPECTRUM, 100);
+        await vi.waitUntil(() => commit.mock.calls.length >= 2, {
+            interval: 100,
+            timeout: 6000
+        })
```

Comment:
This code block is largely repeated 5 times - could pull it into a separate util method, especially is shared time values might need tweaking?
...or maybe just the waitUntil bit, which also shows up in other modules?..

Reply:
I did try this, was running into issues and i am not sure why honestly, but it seems to work when i have vi.waitUntil in the test function, as a compromise I have abstracted out the interval and timeout in the test suites that they were used multiple times in!